components/script/dom/location.rs

```diff
@@ -210,8 +261,19 @@ impl LocationMethods for Location {
 
     // https://html.spec.whatwg.org/multipage/#dom-location-pathname
     fn SetPathname(&self, value: USVString) -> ErrorResult {
-        self.check_same_origin_domain()?;
-        self.set_url_component(value, UrlHelper::SetPathname);
+        // Step 1: If this Location object's relevant Document is null, then return.
+        if self.window.has_document() {
+            // Step 2: If this Location object's relevant Document's origin is not
+            // same origin-domain with the entry settings object's origin, then
+            // throw a "SecurityError" DOMException.
+            self.check_same_origin_domain()?;
+            // Step 3: Let copyURL be a copy of this Location object's url.
+            let copy_url = self.get_url();
+            // Step 4: If copyURL's cannot-be-a-base-URL flag is set, terminate these steps.
```

Comment:
Are we doing step 5 currently? If not, I think we can do it using `as_mut_url().set_path`.

Reply:
Oh, I assumed we were executing those steps as a part of `url::quirks::set_path` through `UrlHelper::SetPathname`, but those steps seem to correspond to dom url spec.